src/main/java/fruitshop/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package fruitshop.strategy;
+
+import fruitshop.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler>
+                                         operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void getOperation(FruitTransaction fruitTransaction) {
+        OperationHandler operationHandler = operationHandlers.get(fruitTransaction.getOperation());
+        operationHandler.handler(fruitTransaction);
```

Comment:
It's possible for `operationHandlers.get(fruitTransaction.getOperation())` to return `null` if the operation is not found in the map. This would lead to a `NullPointerException` when `handler` is called on `null`. You should add a null check and throw a custom exception with an informative message if no handler is found for the given operation.